### ISSUE TITLE
Writer freezes permanently upon closing an opened document

### ISSUE THREAD

**Issue by darkseji** [writingtool-org/writingtool#17]
I've already submitted a bug for LO here https://bugs.documentfoundation.org/show_bug.cgi?id=163965 and they are suspecting Skia issues. However, I've been experimenting a bit and there might be a connection to the WritingTool extension.

In general, Writer freezes when closing a document and requires to be killed from a task manager panel. Usually this is followed by a document recovery dialog/crash report, but not always. I've never seen this behavior with an empty/new document, only with something opened first - one document or many. Doesn't matter how long the document stays open, it can happen after closing it in 1 minute or after 5 hours.

The issue seems to disappear when I disable Skia. However, I've tried out a fresh new profile and have not experienced any issues until I've installed the WritingTool extension - at that moment with Skia enabled the freezes came back. I'm not able to say how this is connected to Skia and if it's not just a correlation but maybe there are ways I could investigate from the extension point of view?

**Comment by FredKruse** [comment]
I hope, I solved the problem. Please test it with the nightly tomorrow.

**Comment by darkseji** [comment]
Thanks! I'll be away for a couple of days so I'll be able to check next week. Will report back if I experience any issues.

**Comment by darkseji** [comment]
@FredKruse I thing things are better now. My impression is that Writer initializes/opens a document a lot faster plus I was not really able to recreate the behavior I had described. I think 9 of 10 times the document closes without issues.

That 1 ot of 10 times, however, is that after closing the last document the LO process stays live and visible in the task manager and needs to be killed manually. I can't say if this is LO related or extension related. Same thing with opening and quickly closing multiple documents that crashes Writer without any error message - this might be LO internal thing.

Anyway, I see definitive improvement. I'll be observing things while working on different docs and let you know if the issue appears again.

**Comment by darkseji** [comment]
So, my Writer just froze. Had two files open concurrently - one for a couple of hours, the other one for about 5 mins, tried to close that last one. Apparently the issue is still there but somehow it's less frequent.

**Comment by FredKruse** [comment]
The problem is that I have no other idea what the problem could be. WritingTool does not give any errors. It is a Java program that should display errors directly as a message box, or at least write them to the log file. I have not found anything like that in my tests.
The last time I revised this point, I looked through all the functions that are relevant when closing a document and revised everything that, I thought, could cause problems.
I now believe that it is a LibreOffice problem. Perhaps it has something to do with the interfaces that WritingTool uses. Even if something goes wrong in WritingTool, the interfaces should be robust enough that the entire word-processing program does not crash. Have you reported the problem to LibreOffice?

**Comment by darkseji** [comment]
Yes, I've reported it some time ago, they think it's a Skia-related issue: https://bugs.documentfoundation.org/show_bug.cgi?id=163965 so that might be my gfx card or virtually anything else at my end as I guess I might be the only one with that problem. 

Anyway, things are a bit better now. Thank you for your work and for keeping the extension alive for LO. :)

**Comment by FredKruse** [comment]
Is the problem solved? Can I close the issue?

**Comment by darkseji** [comment]
Yes, sure.
